src/containers/pages/OrganizationViews/AssignPractitioners/index.tsx

```diff
@@ -0,0 +1,279 @@
+  ORGANIZATIONS_LABEL,
+  ORGANIZATIONS_LIST_URL,
+  PRACTITIONER_CODE,
+  PRACTITIONERS,
+  TO,
+} from '../../../../constants';
+import { generateNameSpacedUUID } from '../../../../helpers/utils';
+import { OpenSRPService } from '../../../../services/opensrp';
+import organizationsReducer, {
+  fetchOrganizations,
+  getOrganizationById,
+  Organization,
+  reducerName as organizationReducerName,
+} from '../../../../store/ducks/opensrp/organizations';
+import practitionersReducer, {
+  fetchPractitionerRoles,
+  getPractitionersByOrgId,
+  Practitioner,
+  reducerName as practitionerReducerName,
+} from '../../../../store/ducks/opensrp/practitioners';
+import { loadOrganization, loadOrgPractitioners } from '../serviceHooks';
+import { filterOptions, formatOptions, SelectOption, styles } from './utils';
+
+reducerRegistry.register(organizationReducerName, organizationsReducer);
+reducerRegistry.register(practitionerReducerName, practitionersReducer);
+
+/** Props for AssignPractitioner component */
+interface AssignPractitionerProps {
+  fetchOrganizationsCreator: typeof fetchOrganizations;
+  fetchPractitionerRolesCreator: typeof fetchPractitionerRoles;
+  organization: Organization | null;
+  serviceClass: typeof OpenSRPService;
+  assignedPractitioners: Practitioner[];
+}
+
+/** default props for AssignPractitioner component */
+const defaultAssignPractitionerProps: AssignPractitionerProps = {
+  assignedPractitioners: [],
+  fetchOrganizationsCreator: fetchOrganizations,
+  fetchPractitionerRolesCreator: fetchPractitionerRoles,
+  organization: null,
+  serviceClass: OpenSRPService,
+};
+
+/** type intersection for all types that pertain to the props */
+export type PropsTypes = AssignPractitionerProps & RouteComponentProps<RouteParams>;
+
+/** AssignPractitioner component */
+const AssignPractitioner: React.FC<PropsTypes> = props => {
+  const {
+    serviceClass,
+    fetchOrganizationsCreator,
+    organization,
+    fetchPractitionerRolesCreator,
+    assignedPractitioners,
+  } = props;
+  const [selectedOptions, setSelectedOptions] = useState<OptionsType<SelectOption>>([]);
+
+  useEffect(() => {
+    const organizationId = props.match.params.id;
+    loadOrganization(organizationId, serviceClass, fetchOrganizationsCreator);
+    loadOrgPractitioners(organizationId, serviceClass, fetchPractitionerRolesCreator);
+  }, []);
+
+  if (!organization) {
+    return <Loading />;
+  }
+
+  // TODO - scenario where this request doesn't return all practitioners in a single query
+  /** load all practitioners at least all of those returned
+   * in a single call to Practitioners endpoint
+   * @param {typeof OpenSRPService} service - the OpenSRPService
+   */
+  const loadAllPractitioners = async (service: typeof OpenSRPService = OpenSRPService) => {
+    const serve = new service(OPENSRP_PRACTITIONER_ENDPOINT);
+    const allPractitioners = await serve.list();
+    return formatOptions(allPractitioners);
+  };
```

Comment:
Yes i think it can, I was hesitant due to the fact that there is no way to tell opensrp for this endpoint how many records should be returned, so i have to trust that the default max page size for this endpoint does not change.

Reply:
No actually on second thought we would need some sought of pagination on this endpoint

Reply:
will create a ticket for this

Reply:
#535